### ISSUE TITLE
Flood (experimental) not working with attribute set by the wizard

### ISSUE THREAD

**Issue by vdeparday** [inasafe/inasafe#1158]
Very similar to bug #1156, I get an error when trying to run a flood (experimental) impact function after setting up the hazard layer with the wizard or even if I add an attribute called  affected with a value equal to 1.

Full report from InaSAFE:

```
Processing started

Please wait - processing may take a while depending on your hardware configuration and the analysis extents and data.

This analysis will calculate the impact of hti_hazardhydrology_rareflood_westdepartment_nathat_polygon_052Polygon on inasafe_download_urbuildings and bullet_list the results aggregated by hti_boundaries_sections_communales_adm3_cnigs_polygon .

The following postprocessors will be used:

Building type: Calculates building types related statistics.
Checking available memory

Computed cellsize was None. Memory check currently only works for raster input layers.
Select attribute

Please select which attribute you want to use as ID for the aggregated results

Preparing aggregation layer

We are clipping the aggregation layer to match the intersection of the hazard and exposure layer extents.

Calculating impact

This may take a little while - we are computing the areas that will be impacted by the hazard and writing the result to a new layer.

Problem

The following problem(s) were encountered whilst running the analysis.

GetDataError
Suggestion

You can try the following to resolve the issue:

There are no objects in the hazard layer with "Affected value"='1.0'. Please check the value or use other extent.
Details

These additional details were reported when the problem occurred.

An exception occurred when calculating the results. Calculation error encountered:
There are no objects in the hazard layer with "Affected value"='1.0'. Please check the value or use other extent.
Diagnostics (click for details)

In file "/Users/viv/.qgis2/python/plugins/inasafe/safe_qgis/utilities/impact_calculator_thread.py", line 186, in run check_integrity=self._check_integrity)
In file "/Users/viv/.qgis2/python/plugins/inasafe/safe_qgis/safe_interface.py", line 537, in calculate_safe_impact check_integrity=check_integrity)
In file "/Users/viv/.qgis2/python/plugins/inasafe/safe/engine/core.py", line 68, in calculate_impact F = impact_function.run(layers)
In file "/Users/viv/.qgis2/python/plugins/inasafe/safe/impact_functions/inundation/flood_building_impact_qgis.py", line 212, in run raise GetDataError(message)
```

**Comment by ismailsunni** [comment]
Need to fix #2281 first, since the layers are not in EPSG4326, and it will cause errors.

**Comment by ismailsunni** [comment]
Move to version 3.3 since we need to fix #2296 and #2293

**Comment by timlinux** [comment]
@vdepardaysorry I know it is long after the fact, but are you able to attach some sample data that we can use to replicate this?

**Comment by timlinux** [comment]
This should be resolved in 4.0 - closing
